File: timeplan/front/src/app/models/student.ts
import {Group} from "./group";

export class Student{
  choosen:boolean = false;
  idstudent:number;
  fname: string;
  lname:string;
  email:string;
  password: string;
  role:string;
  group:Group;

  constructor(){}


  static cloneStudent(student: Student):Student{
    let copyStudent = new Student();
    copyStudent.idstudent = student.idstudent;
    copyStudent.fname = student.fname;
    copyStudent.lname =  student.lname;
    copyStudent.email =   student.email;
    copyStudent.password =  student.password;
    copyStudent.group = student.group;
    return copyStudent;
  }
}

File: timeplan/be/src/main/java/com/netcracker/be/entity/Student.java
package com.netcracker.be.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import javax.persistence.*;

@Entity
@Table(name ="studenttest")
public class Student extends  User{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name ="idstudent")
    private Long idstudent;
    @Column(name="fname")
    private String fname;
    @Column(name="lname")
    private String lname;
    @Column(name="email")
    private String email;
    @Column(name="password")
    private String password;
    @ManyToOne(fetch = FetchType.EAGER, cascade = {CascadeType.REFRESH, CascadeType.MERGE})
    @JoinColumn(name = "idgroup")
    //@JsonIgnore
    private Group group;
    private String role;

    @Override
    public String getRole() {
        return role;
    }

    @Override
    public void setRole(String role) {
        this.role = role;
    }

    public Long getIdstudent() {
        return idstudent;
    }

    public void setIdstudent(Long idstudent) {
        this.idstudent = idstudent;
    }

    public String getFname() {
        return fname;
    }

    public void setFname(String fname) {
        this.fname = fname;
    }

    public String getLname() {
        return lname;
    }

    public void setLname(String lname) {
        this.lname = lname;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Group getGroup() {
        return group;
    }

    public void setGroup(Group group) {
        this.group = group;
    }
}
